### PR TITLE
User income dev

### PR DESCRIPTION
**Pull request by Tordek947**

### CODE REVIEW

File: src/main/java/com/demo/entrepreneur/dto/user/RequestUserDto.java [hunk 1]
```diff
@@ -18,19 +21,22 @@
...
     private String password;
 
-    @Schema(required = true)
-    @JsonProperty("email")
+    @JsonProperty(value = "email", required = true)
     private String email;
 
-    @Schema(required = true)
-    @JsonProperty("firstName")
+    @JsonProperty(value = "firstName", required = false)
```
Comment (Zephyrkina) [inline src/main/java/com/demo/entrepreneur/dto/user/RequestUserDto.java]: Do we need to write "required = false" property? This option is used by default

File: src/main/java/com/demo/entrepreneur/entity/User.java [hunk 2]
```diff
@@ -1,30 +1,46 @@
...
-    @Column(name = "email")
+    @Column(name = "email", nullable = false)
     private String email;
     @Column(name = "firstname")
     private String firstName;
     @Column(name = "lastname")
     private String lastName;
+    @Column(name = "apprx_income_usd", nullable = false)
+    private Integer approximateIncomeInUsd;
+    @Column(name = "taxGroup", nullable = false)
```
Comment (Zephyrkina) [inline src/main/java/com/demo/entrepreneur/entity/User.java]: Are you sure about correctness of this column name? 
1) it seems that we used underlines for column naming;
2) in changelog there is "tax_group" column

File: src/main/java/com/demo/entrepreneur/dto/user/RequestUserDto.java (hunk 1 above)
Comment (Tordek947) [reply src/main/java/com/demo/entrepreneur/dto/user/RequestUserDto.java]: You are completely right, made it for clearness before previous meeting. I'm going to remove it, thanks.

File: src/main/java/com/demo/entrepreneur/entity/User.java (hunk 2 above)
Comment (Tordek947) [reply src/main/java/com/demo/entrepreneur/entity/User.java]: It seems like my typo. Going to recheck it and correct if it is necessary. Thank you.

File: src/main/java/com/demo/entrepreneur/dto/user/RequestUserDto.java (hunk 1 above)
Comment (Tordek947) [reply src/main/java/com/demo/entrepreneur/dto/user/RequestUserDto.java]: Done

File: src/main/java/com/demo/entrepreneur/entity/User.java (hunk 2 above)
Comment (Tordek947) [reply src/main/java/com/demo/entrepreneur/entity/User.java]: Corrected

File: src/main/java/com/demo/entrepreneur/enumeration/TaxGroup.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    III(3, true),
+    IV(4, false),
+    V(5, true);
+
+    private final Integer groupNumber;
+    private final boolean supported;
+
+    public static TaxGroup valueOfGroupNumber(Integer groupNumber) {
+        return Stream.of(values()).filter(value -> value.getGroupNumber().equals(groupNumber)).findAny().orElseThrow(
+                () -> new RuntimeException(String.format("TaxGroup value for groupNumber %s not found", groupNumber)));
```
Comment (Zephyrkina) [inline src/main/java/com/demo/entrepreneur/enumeration/TaxGroup.java]: What if make this part of code more readable? Something like following:
` return Stream.of(values())
                .filter(value -> value.getGroupNumber().equals(groupNumber))
                .findAny()
                ...
`

File: src/main/java/com/demo/entrepreneur/mapping/mapper/impl/ResponseUserMapper.java [hunk 4]
```diff
@@ -3,18 +3,26 @@
...
-        return ResponseUserDto.builder()// @formatter:off
+        ResponseUserDto data = ResponseUserDto.builder()// @formatter:off
                 .email(entity.getEmail())
                 .firstName(entity.getFirstName())
                 .lastName(entity.getLastName())
-                .login(entity.getLogin()).build();// @formatter:on
+                .login(entity.getLogin())
+                .approximateIncomeInUsd(entity.getApproximateIncomeInUsd())
+                .taxGroup(entity.getTaxGroup().getGroupNumber()).build();// @formatter:on
+        log.info(data.toString());
```
Comment (Zephyrkina) [inline src/main/java/com/demo/entrepreneur/mapping/mapper/impl/ResponseUserMapper.java]: Do we really need logging here?

Comment (Tordek947) [reply src/main/java/com/demo/entrepreneur/mapping/mapper/impl/ResponseUserMapper.java]: Going to remove it now, I used this line for debug purposes only

File: src/main/java/com/demo/entrepreneur/enumeration/TaxGroup.java (hunk 3 above)
Comment (Tordek947) [reply src/main/java/com/demo/entrepreneur/enumeration/TaxGroup.java]: As I am fixing the logging issue anyway, will apply this changes also.

File: src/main/java/com/demo/entrepreneur/entity/TaxGroup.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+import lombok.Getter;
+
+@Getter
+public enum TaxGroup {
+    GENERAL(0, false), I(1, false), II(2, false), III(3, true), IV(4, false), V(5, true), UNDEFINED(-1, false);
+
+    private TaxGroup(Integer groupNumber, boolean supported) {
+        this.groupNumber = groupNumber;
+        this.supported = supported;
```
[the remark this answers is not in the document]
Comment (Tordek947) [reply src/main/java/com/demo/entrepreneur/entity/TaxGroup.java]: Thank you, @VladRybkin , I will remove this modifier for the next pull request.

**Pull request merged**